src/Components/Components/src/EventCallbackFactoryBinderExtensions.cs

```diff
@@ -2,13 +2,27 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Concurrent;
+using System.ComponentModel;
+using System.Diagnostics;
 using System.Globalization;
+using System.Reflection;
 
 namespace Microsoft.AspNetCore.Components
 {
     /// <summary>
     /// Contains extension methods for two-way binding using <see cref="EventCallback"/>. For internal use only.
     /// </summary>
+    //
+    // NOTE: for number parsing, the HTML5 spec dictates that <input type="number"> the DOM will represent
+    // number values as floating point numbers using `.` as the period separator. This is NOT culture senstive.
+    // Put another way, the user might see `,` as their decimal separator, but the value available in events
+    // to JS code is always simpilar to what .NET parses with InvariantCulture.
+    //
+    // See: https://www.w3.org/TR/html5/sec-forms.html#number-state-typenumber
+    // See: https://www.w3.org/TR/html5/infrastructure.html#valid-floating-point-number
+    //
+    // For now we're not necessarily handling this correctly since we parse the same way for number and text.
```

Comment:
This is a comment I had written about the current state of globalization and our conversions before I impleemented type converter support. I want to leave it in because it's good info to keep close to the code.